service/raft/raft_address_map.hh

```diff
@@ -167,13 +173,90 @@ class raft_address_map : public peering_sharded_service<raft_address_map<Clock>>
         }
     }
 
+    template <std::invocable<raft_address_map&> F>
+    void replicate(F f, std::experimental::source_location l = std::experimental::source_location::current()) {
+        if (_gate.is_closed()) {
+            return;
+        }
+
+        _replication_fiber = _replication_fiber.then([this, f = std::move(f), l] () -> future<> {
```

Comment:
So the replication is async and other shards may learn about the new address after some time. Can this cause us problems? Should other shards be eager in asking shard0 about non existing mapping?

Reply:
> So the replication is async and other shards may learn about the new address after some time. Can this cause us problems? 

Lacking an entry in raft_address_map is equivalent to a temporary network unavailability, and both intended use cases (failure detector, raft servers) tolerate those with no problems.

That said, it's most likely that the replication message will reach the other shard before the raft entry needs to be accessed. For failure detector for example, if we `set` the entry in the map first and add it to the failure detector second, the message in raft address map will most likely reach the other shard before the message in failure detector that spawns the pinging worker on the other shard. For future raft groups running on other shards, I expect that all servers will be members of group 0 before their server IDs are used in other groups.

> Should other shards be eager in asking shard0 about non existing mapping?

That would require changing the interface of `find` to return `future<>`, which I think we shouldn't do unless we're convinced it's absolutely necessary. Also it wouldn't fix the problem of having outdated mapping (Kostja is adding the possiblity of mappings to change).